File: leetcode/58.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:

        word_len = 0
        flag = 0
        for idx in range(len(s)-1,-1,-1):
            if s[idx] != ' ':
                flag = 1
                word_len +=1
            if flag and s[idx] == ' ':
                break
        
        return word_len


s = " "

sol = Solution()
word_len = sol.lengthOfLastWord(s)
print(word_len)

File: leetcode/arrays/58.go
package main

import "fmt"

func lengthOfLastWord(s string) int {
    ans, flag := 0, 0

	for idx := len(s)-1;idx>=0;idx--{
		if rune(s[idx]) != ' '{
			flag = 1
		} 
		
		if flag ==1 && rune(s[idx])==' '{
			return ans
		} else if flag ==1{
			ans++
		}

	}
	return ans
}

func main() {

	s := "H HH abc "

	ans := lengthOfLastWord(s)

	fmt.Println(ans)
}